packages/models/src/metadata.ts

```diff
@@ -52,6 +52,17 @@ export class Metadata<T extends MetadataType> implements Signable {
     this.unrecognizedFields = unrecognizedFields || {};
   }
 
+  public sign(signer: (data: Buffer) => Signature, append = true): void {
+    const bytes = canonicalize(this.signed.toJSON());
+    const signature = signer(bytes);
+
+    if (!append) {
+      this.signatures = {};
+    }
+
+    this.signatures[signature.keyID] = signature;
+  }
```

Comment:
New function to support the generation of signatures for metadata files. Needed by the new `repo-mock` package.